base_app/services/yelp_service.py

```diff
@@ -4,33 +4,49 @@
 import requests
 from base_app.models.restaurant import Restaurant
 from base_app.models.restaurant_schema import RestaurantSchema
+from base_app.services.google_service import GoogleService
 from dotenv import load_dotenv
 load_dotenv()
 
 class YelpService:
 
     def get_recommendation(self, params):
         restaurant_id = self.get_restaurant(params)
-        url = 'https://api.yelp.com/v3/businesses/{}'.format(restaurant_id)
-        response = self.connection(url)
-        json_data = json.dumps(response.json())
-        recommendation = Restaurant.from_json(json_data)
-        schema = RestaurantSchema()
-        result = schema.dump(recommendation)
-        return result
+        if restaurant_id:
+            url = 'https://api.yelp.com/v3/businesses/{}'.format(restaurant_id)
+            response = self.connection(url)
+            json_data = json.dumps(response.json())
+            if 'coordinates' in response.json().keys():
+                google_service = GoogleService()
+                website = google_service.get_website(json_data)
+            elif 'url' in response.json().keys(): # Will return the Restaurant's Yelp Business Page if no Coordinates available
+                website = response.json()['url']
+            recommendation = Restaurant.from_json(json_data, website)
+            schema = RestaurantSchema()
+            result = schema.dump(recommendation)
+            return result
+        elif 'radius' in params.keys() and params['radius'] == 16000:
+            params.pop('radius', None)
+            self.get_recommendation(params)
```

Comment:
Does this reference the walk option?

Reply:
Line 30 does. The walk option will set the radius to 1,600. If it's set to 16000 then it's either default or Drive.